typescript/utils/src/objects.ts

```diff
@@ -208,3 +209,99 @@ export function stringifyObject(
   }
   return yamlStringify(JSON.parse(json), null, space);
 }
+
+interface ObjectDiffOutput {
+  actual: any;
+  expected: any;
+}
+
+export type ObjectDiff =
+  | {
+      [key: string]: ObjectDiffOutput | ObjectDiff;
+    }
+  | ObjectDiff[];
+
+/**
+ * Merges 2 objects showing any difference in value for common fields.
+ */
+export function diffObjMerge(
+  actual: Record<string, any>,
+  expected: Record<string, any>,
+  max_depth = 10,
+): {
+  mergedObject: ObjectDiff;
+  isInvalid: boolean;
+} {
+  if (max_depth === 0) {
+    throw new Error('diffObjMerge tried to go too deep');
+  }
+
+  let isDiff = false;
+  if (!isObject(actual) && !isObject(expected) && actual === expected) {
```

Comment:
super edge case -- for some number values such as `totalSupply`, this may change by the time the config is derived. so you might get a false positive sometimes.

think this is acceptable, but up to you if you want to address it :)